internal/checks/latency.go

```diff
@@ -0,0 +1,484 @@
+package checks
+
+import (
+	"context"
+	"math"
+	"net/http"
+	"strconv"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/failsafe-go/failsafe-go/circuitbreaker"
+
+	"github.com/satsuma-data/node-gateway/internal/client"
+	conf "github.com/satsuma-data/node-gateway/internal/config"
+	"github.com/satsuma-data/node-gateway/internal/metrics"
+	"github.com/satsuma-data/node-gateway/internal/types"
+	"go.uber.org/zap"
+)
+
+const (
+	ResponseCodeWildcard  = 'x'
```

Comment:
My bad, the usage of "x" as a wildcard in example in the design doc wasn't meant to be interpreted literally 😓 it was more of a demonstration that wildcards should be supported in some fashion.

WDYT about just making the responsecodes/errorstrings as pure regex? That way it's easier for users since regex is pretty standard. Regex also allows more flexibility.

Reply:
SGTM!